src/macaron/slsa_analyzer/analyzer.py

```diff
@@ -170,17 +170,15 @@ def run(self, user_config: dict, sbom_path: str = "", skip_deps: bool = False) -
                     logger.info("Start analyzing the dependencies.")
                     for config in deps_config:
                         dep_status: SCMStatus = config.get_value("available")
-                        if dep_status != SCMStatus.AVAILABLE:
+                        if dep_status == SCMStatus.DUPLICATED_SCM:
```

Comment:
With this change, if a repository is missing, we will no longer report `MISSING REPO URL` and run the analysis anyway, reporting that result is available
![image](https://github.com/oracle/macaron/assets/27933054/17e48645-c3dd-4bc5-bfdc-5ea1a5434710)

While the HTML report for the dependency reports that the URL is missing
![image](https://github.com/oracle/macaron/assets/27933054/a46a7d3d-3798-4a09-a7a4-24b6c84a4463)

Is that intended?

Reply:
Yes this is intended. I think I made it this way is because the data model allows a software component to have no repository. This would mean that if a software component doesn't have any repository available:
- All checks should still be run (even though they will all be failed because we only have repository-based checks for now).
- A [Component](https://github.com/oracle/macaron/blob/4282c6c6e5ffc080a094fec7622b6d07eedd8987/src/macaron/database/table_definitions.py#L120) instance is still created with its Repository field set to empty. 
Because we still create a software component and perform the analysis (even though all checks will fail) for the target software component (as agreed for this feature), I figure we should keep it consistent with the dependencies as well. Therefore, we don't skip dependency software component if they don't have a repository.